Subject: [PATCH] optimized: 优化bigtop的代理。解决spark 3.2.3 在其他镜像源没有的问题
---
Index: bigtop.bom
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop.bom b/bigtop.bom
--- a/bigtop.bom	(revision b5254cf1a02f2270ccdfb560b82d733138754044)
+++ b/bigtop.bom	(revision 4a16750e89141e6f11ce433a8a06f85d76976501)
@@ -101,7 +101,7 @@
         'scala' { version = '2.12.13'; version_base = version }
     }
     apache {
-        APACHE_MIRROR = "https://mirrors.tuna.tsinghua.edu.cn/apache"
+        APACHE_MIRROR = "https://mirrors.huaweicloud.com/apache"
         APACHE_ARCHIVE = "https://mirrors.huaweicloud.com/apache"
     }
 /** End of Base Configuration */
