Subject: [PATCH] feature: 支持ozone 1.4.1 下载
---
Index: bigtop.bom
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop.bom b/bigtop.bom
--- a/bigtop.bom	(revision dcc7804c741d712ffcff8a998aad1eed2636eaf5)
+++ b/bigtop.bom	(revision 615890dfb63f9ab9ffc51557421623f90e43931d)
@@ -614,5 +614,20 @@
                 archive = site
             }
         }
+
+        'ozone' {
+            name = 'ozone'
+            rpm_pkg_suffix = "_" + bigtop.base_version.replace(".", "_")
+            relNotes = 'ozone v1.4.1'
+            version { base = '1.4.1'; pkg = base; release = 1 }
+            tarball {
+                source = "${name}-${version.base}.tar.gz"
+                destination = source
+            }
+            url {
+                site = bigtop.gh_proxy + "https://github.com/apache/ozone/archive/refs/tags"
+                archive = site
+            }
+        }
     }
 }
