Subject: [PATCH] feature: 支持celeborn 0.5.3
---
Index: bigtop.bom
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop.bom b/bigtop.bom
--- a/bigtop.bom	(revision e249c693e58335f0c6c34e40502bc4950201d002)
+++ b/bigtop.bom	(date 1738895730573)
@@ -599,5 +599,20 @@
                 archive = site
             }
         }
+
+        'celeborn' {
+            name = 'celeborn'
+            rpm_pkg_suffix = "_" + bigtop.base_version.replace(".", "_")
+            relNotes = 'celeborn v0.5.3'
+            version { base = '0.5.3'; pkg = base; release = 1 }
+            tarball {
+                source = "v${version.base}.tar.gz"
+                destination = "${name}-${version.base}.tar.gz"
+            }
+            url {
+                site = bigtop.gh_proxy + "https://github.com/apache/celeborn/archive/refs/tags"
+                archive = site
+            }
+        }
     }
 }
