Subject: [PATCH] feature: 尝试支持categraf
feature: 尝试支持 nightingale
---
Index: bigtop.bom
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop.bom b/bigtop.bom
--- a/bigtop.bom	(revision 6d495081201e156c8a253a4375a203791beacbe4)
+++ b/bigtop.bom	(revision 5789ef433ad932f250c8d8c4cf0038d466b8c7e8)
@@ -539,5 +539,35 @@
                 archive = site
             }
         }
+
+        'nightingale' {
+            name = 'nightingale'
+            rpm_pkg_suffix = "_" + bigtop.base_version.replace(".", "_")
+            relNotes = 'Nightingale v7.7.2'
+            version { base = '7.7.2'; pkg = base; release = 1 }
+            tarball {
+                source = "n9e-v${version.base}-linux-amd64.tar.gz"
+                destination = source
+            }
+            url {
+                site = bigtop.gh_proxy + "https://github.com/ccfos/nightingale/releases/download/v${version.base}"
+                archive = site
+            }
+        }
+
+        'categraf' {
+            name = 'categraf'
+            rpm_pkg_suffix = "_" + bigtop.base_version.replace(".", "_")
+            relNotes = 'Categraf v0.4.1'
+            version { base = '0.4.1'; pkg = base; release = 1 }
+            tarball {
+                source = "categraf-v${version.base}-linux-amd64.tar.gz"
+                destination = source
+            }
+            url {
+                site = bigtop.gh_proxy + "https://github.com/flashcatcloud/categraf/releases/download/v${version.base}"
+                archive = site
+            }
+        }
     }
 }
