Subject: [PATCH] fixed: 修复redis 配置文件错位问题
---
Index: ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/params_linux.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/params_linux.py b/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/params_linux.py
--- a/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/params_linux.py	(revision 4178a369d16ac52a4daccafc3e87b206d88348e3)
+++ b/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/params_linux.py	(revision fc29a1d627a3dd751e16e3c4c023d7e0461fd861)
@@ -17,7 +17,7 @@
 stack_root = Script.get_stack_root()

 # parent
-redis_conf_parent = '/etc/redis'
+redis_conf_parent = '/etc/redis/conf'

 # bin dir
 master_bin = format("{stack_root}/current/{master_component_directory}/src")
Index: ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/redis_master.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/redis_master.py b/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/redis_master.py
--- a/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/redis_master.py	(revision 4178a369d16ac52a4daccafc3e87b206d88348e3)
+++ b/ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/REDIS/package/scripts/redis_master.py	(revision fc29a1d627a3dd751e16e3c4c023d7e0461fd861)
@@ -1,6 +1,6 @@
 # coding=utf-8
+import os
 import time
-
 from resource_management import Template
 from resource_management.core.logger import Logger
 from resource_management.core.resources.system import Execute, File, Directory
@@ -27,11 +27,12 @@
                   create_parents=True)

         # 生成 redis.conf 配置文件
-        File("/etc/redis/redis.conf",
-             content=Template("redis.conf.j2"),
-             owner=params.redis_user,
-             group=params.user_group,
-             mode=0644)
+        File(
+            os.path.join(params.redis_conf_parent, "redis.conf"),
+            content=Template("redis.conf.j2"),
+            owner=params.redis_user,
+            group=params.user_group,
+            mode=0644)

     def start(self, env):
         import params
